### ISSUE TITLE
Porting to Pytorch 2.0

### ISSUE THREAD

**Issue by Geeks-Sid** [mlcommons/GaNDLF#587]
**Is your feature request related to a problem? Please describe.**
"As PyTorch 2.0 is approaching its release and promising significant benefits, particularly in model compilation, it would be beneficial for GaNDLF to migrate to the platform once it becomes stable. To learn more about PyTorch 2.0, visit [here](https://pytorch.org/get-started/pytorch-2.0/).

**Describe the solution you'd like**
A transition after *tagging* GaNDLF to move to pytorch 2.0

**Describe alternatives you've considered**
-

**Additional context**
-

**Comment by szmazurek** [comment]
Yeah, taking care of that -  this upgrade and the related ones will be out in the next weeks.
